File: python/Sorting/insertion_sort.py
n = int(input("Enter number of elements: "))

arr = list(map(int, input(f"Enter {n} elements (space separated): ").split()))

for i in range(1, n):
    j = i
    while (j > 0 and arr[j - 1] > arr[j]):
        arr[j - 1], arr[j] = arr[j], arr[j - 1]
        j -= 1
    
print("Sorted array in insertion_sort", arr)

File: java/Sorting/InSertion_Sort.java
package Sorting;
import java.util.*;

public class InSertion_Sort {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter number of elements: ");
        int n = sc.nextInt();

        int[] arr = new int[n];
        System.out.print("Enter " + n + " elements:");

        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        for (int i = 0; i <= n - 1; i++) {
            int j = i;
            while(j > 0 && arr[j - 1] > arr[j]) {
                int temp = arr[j-1];
                arr[j - 1] = arr[j];
                arr[j] = temp;
                j--;
            }
        }

        System.out.println("Sorted arrays in insertion_sort: " + Arrays.toString(arr));
        sc.close();
    }
}
